plugins/inputs/openweathermap/openweathermap.go

```diff
@@ -293,6 +299,9 @@ func gatherForecast(acc telegraf.Accumulator, status *Status) {
 			tags["condition_main"] = e.Weather[0].Main
 		}
 		tags["forecast"] = fmt.Sprintf("%dh", (i+1)*3)
+		if timestamp == "prediction" {
+			tm = tm.Add(time.Duration((i+1)*3) * time.Hour)
+		}
```

Comment:
Hmm I think this won't work as you accumulate the time. I think in case you assign it to the same variable you should do
```suggestion
		if timestamp == "prediction" {
			tm = tm.Add(3 * time.Hour)
		}
```
here, shouldn't you?